### DISCUSSION TITLE
Decided: MeltanoLabs for more than just Singer Plugins

### DISCUSSION THREAD

**Discussion by pnadolny13** [MeltanoLabs/Meta#15] [xArchived]
### Decision - 2022-06-02

MeltanoLabs is the place for Meltano community supported projects. More than Singer Plugins can live on MeltanoLabs, but we will not be transferring Meltano Inc supported/maintained projects to MeltanoLabs.

---

I wanted to start a discussion to see what people think about adding more than just Singer taps/targets to MeltanoLabs. The Meltano namespace no longer feels like the best place to store plugins and files related to plugins, primarily right now thats files bundles. The Meltano core team can still be the primary contributors to these packages but having them in MeltanoLabs would better organize the Meltano product separate from plugins that can be used with Meltano.

Now that Meltano is on GitHub it should be trivial to transfer the repo from the Meltano namespace to the MeltanoLabs namespace. Here's a list to give a sense of what we could move:

Singer:
- https://github.com/meltano/tap-stripe
- https://github.com/meltano/tap-salesforce
- https://github.com/meltano/tap-zendesk
- etc.

Files:
- https://github.com/meltano/files-airflow
- https://github.com/meltano/files-dbt-redshift
- https://github.com/meltano/files-dbt-postgres
- https://github.com/meltano/files-gitlab-ci
- etc.

DBT:
- dbt packages like https://gitlab.com/meltano/dbt-tap-gitlab which havent been migrated to GitHub yet

cc @tayloramurphy @aaronsteers

**Comment by tayloramurphy** [comment]
@pnadolny13 I'm very supportive of this. I think it makes complete sense.

@aaronsteers @DouweM @emilylucie thoughts?

**Reply by DouweM** [reply]
@tayloramurphy It doesn't make complete sense to me yet :)

If we are still these projects' primary maintainers, that goes against the “Community-Managed Fork with Community Maintainers” model MeltanoLabs was introduced with in https://meltano.com/blog/launching-meltanolabs-your-home-for-singer-connectors-dbt-packages-and-all-meltano-plugins/. We can choose to redefine the purpose of MeltanoLabs as "where all the Meltano plugins live", no matter who maintains them, but that's a more significant change than just moving over some repos, that I haven't seen a specific argument for yet.

I think the clear distinction between official plugins and community-supported ones is valuable, and would also be surfaced on the Hub (related to https://github.com/meltano/internal-general/issues/205). If someone contributes support for e.g. Dagster and this ends up requiring a file bundle, I think it's fine if that starts out living in someone's personal GitHub namespace, moves to MeltanoLabs if they want to share the maintenance burden with the community, and only moves to the Meltano org once we decide to officially support Dagster at the same level as Airflow dbt, etc.

---

> The Meltano namespace no longer feels like the best place to store plugins and files related to plugins, primarily right now thats files bundles. The Meltano core team can still be the primary contributors to these packages but having them in MeltanoLabs would better organize the Meltano product separate from plugins that can be used with Meltano.

@pnadolny13 Can you explain _why_ the Meltano org no longer feels like the best place, and why it would be good to separate the main Meltano project from the plugins?

**Reply by aaronsteers** [reply]
I think I agree with @DouweM here. I think of MeltanoLabs being 'best effort' community-based support but without guarantees or controls that we need for meltano-managed, meltano-guaranteed components that are more critical and/or internal.

That said, some of the remnants that recently got migrated to `meltano` were actually on our list to phase out entirely. We don't want to continue maintenance on those in GitLab but we also don't necessarily want to promote them to increased visibility in MeltanoLabs.

This very much fits the description for the dbt `transform` bundles, which I'd like to phase out entirely as soon as Meltano or dbt resolves https://github.com/dbt-labs/dbt-core/issues/5302.

**Reply by tayloramurphy** [reply]
@DouweM I think I had a different assumption based on this issue https://github.com/meltano/gitlab-to-github/issues/29 possibly being successful. Given that we can't bring MeltanoLabs into the Enterprise tier then that definitely flips the concern. In my head the move of Meltano to GitLab would make us rethink the framing of MeltanoLabs, but it seems it's not necessary. 

For the scope of this discussion, MeltanoLabs _can_ and _should_ be for more than just Singer plugins, but it won't be for things that Meltano Inc manages and fully supports.

**Reply by DouweM** [reply]
@tayloramurphy Does that mean we can close this discussion as we won't actually be changing anything right now?

**Reply by tayloramurphy** [reply]
@DouweM I think so. I'll update Pat's comment with the decision such that it is.

**Reply by pnadolny13** [reply]
@DouweM @tayloramurphy looks like I'm a little late here but I agree with the decision we landed on. It sounds like the distinction between Meltano org and MeltanoLabs is the support aspect. So we can probably update the readmes in MeltanoLabs to reflect that, accepting any community maintained plugins not just Singer.

**Comment by aaronsteers** [comment]
I think these are all near end of life:

- https://github.com/meltano/tap-stripe
- https://github.com/meltano/tap-salesforce
- https://github.com/meltano/tap-zendesk
- dbt packages like https://gitlab.com/meltano/dbt-tap-gitlab which havent been migrated to GitHub yet


These, conversely, are Meltano-owned, Meltano-guaranteed:

- https://github.com/meltano/files-airflow
- https://github.com/meltano/files-dbt-redshift
- https://github.com/meltano/files-dbt-postgres
- https://github.com/meltano/files-gitlab-ci

I think we made a call a while back that _all_ taps and targets should _either_ be phasing out (the end-of-lifers) or being put up for community ownership. 

Slightly different story for files bundles and such.
